pkg/test/resourcefixture/resourcefixture.go

```diff
@@ -78,50 +78,58 @@ type HeavyFilter func(fixture ResourceFixture) bool
 func LoadWithFilter(t *testing.T, lightFilterFunc LightFilter, heavyFilterFunc HeavyFilter) []ResourceFixture {
 	t.Helper()
 	allCases := make([]ResourceFixture, 0)
-	q := queue.New(1)
-	rootDir := getTestDataPath(t)
-	q.Put(rootDir)
-	for !q.Empty() {
-		items, err := q.Get(1)
+	baseDir := getTestDataPath(t)
+	if err := filepath.WalkDir(baseDir, func(path string, d fs.DirEntry, err error) error {
 		if err != nil {
-			t.Fatalf("error retreiving an item from queue: %v", err)
+			return err
 		}
-		dir := items[0].(string)
-		fileInfos, err := ioutil.ReadDir(dir)
+
+		if !d.IsDir() {
+			return nil
+		}
+
+		// This is a slightly inefficient, but we want to reuse the existing code
+		fileInfos, err := os.ReadDir(path)
 		if err != nil {
-			t.Fatalf("error reading directory '%v': %v", dir, err)
+			return fmt.Errorf("error reading directory '%v': %w", path, err)
 		}
+
 		testToFileName := make(map[string]string)
 		for _, fi := range fileInfos {
 			if fi.IsDir() {
-				q.Put(path.Join(dir, fi.Name()))
 				continue
 			}
 			if !strings.HasSuffix(fi.Name(), ".yaml") {
 				continue
 			}
 			fileNameNoExt := strings.TrimSuffix(fi.Name(), ".yaml")
 			if value, ok := testToFileName[fileNameNoExt]; ok {
-				t.Fatalf("error, conflicting files for test '%v' in '%v': {%v, %v}", fileNameNoExt, dir, value, fi.Name())
```

Comment:
Other fatal calls were converted to return errors for this anonymous function. Why retain the fatal call here?

Reply:
Done